services/core-api/app/api/services/document_manager_service.py

```diff
@@ -19,6 +20,28 @@
 
 class DocumentManagerService():
     document_manager_document_resource_url = f'{Config.DOCUMENT_MANAGER_URL}/documents'
+    
+    @classmethod
+    def validateFileNameAndInitializeFileUploadWithDocumentManager(cls, request, mine, project_guid, document_category):
+
+        metadata = cls._parse_request_metadata(request)
+        file_name = metadata.get('filename')
+        mine_document = MineDocumentSearchUtil.find_by_document_name_and_project_guid(file_name, project_guid)
+
+        if not mine_document: # No existing file found in this application hence continuing the file uploading
+          return DocumentManagerService.initializeFileUploadWithDocumentManager(request, mine, document_category)
+        elif mine_document.is_archived: # An archived file with the same name in this application found, hence responing with 409
+            content = f'{{"description" : "Archived file already exist with the given name: {file_name}", "status_code": 409, \
+                "status": "ARCHIVED_FILE_EXIST", "file_name": "{file_name}", "mine_guid": "{mine_document.mine_guid}", \
+                    "file_type": "{mine_document.document_class}", "update_timestamp": "{mine_document.update_timestamp}", "update_user": "{mine_document.update_user}", \
+                        "mine_document_guid": "{mine_document.mine_document_guid}"}}'
+            return Response(content, 409)
+        else: # The found file with the same name in this application is not archived.
+            content = f'{{"description" : "File already exist with the given name: {file_name}. Replace with the new version", \
+                "status_code": 409, "status": "REPLACEABLE_FILE_EXIST", "file_name": "{file_name}", "mine_guid": "{mine_document.mine_guid}", \
+                    "file_type": "{mine_document.document_class}", "update_timestamp": "{mine_document.update_timestamp}", "update_user": "{mine_document.update_user}", \
+                        "mine_document_guid": "{mine_document.mine_document_guid}"}}'
```

Comment:
These are a bit difficult to read, would it be possible to maybe define them as a json obj and use `json.dumps` to convert them to a string?

Reply:
Updated accordingly.